crates/rolldown_plugin_json/src/lib.rs

```diff
@@ -92,9 +106,60 @@ fn is_special_query(ext: &str) -> bool {
   false
 }
 
+fn get_forbidden_identifiers() -> &'static HashSet<&'static str> {
+  static FORBIDDEN_IDENTIFIERS: OnceLock<HashSet<&'static str>> = OnceLock::new();
+  FORBIDDEN_IDENTIFIERS.get_or_init(|| {
+    let mut set: HashSet<&'static str> =
+      RESERVED_WORDS.split_whitespace().chain(BUILTINS.split_whitespace()).collect();
+    set.insert("");
+    set
+  })
+}
+
+fn make_legal_identifier(ident: &str) -> String {
```

Comment:
We have implementation here https://github.com/rolldown/rolldown/blob/d116b6be242be48fe23ccd3c0fc8e0c17bced3a5/crates/rolldown_utils/src/ecma_script.rs#L12-L12

Reply:
And you don't need to  care about the `reserved` keywords, it will be automatically deconflict in the finalized phase.

Reply:
<img width="771" alt="image" src="https://github.com/user-attachments/assets/652263c9-f94c-4638-bf1d-c370a7a57c08">

I used legitimize_identifier_name to check whether an identifier is valid. However, for identifiers like const or true, the function returns the same identifier, which causes the parser to fail when parsing the code. Did I miss something?

Reply:
My bad, this is phase before parsing, so you need to convert it by hand.
you could use  https://github.com/rolldown/rolldown/blob/34cd0dc8c658946a12c614a1558433bce6766070/crates/rolldown_utils/src/ecma_script.rs#L8-L8 function, to check, this should be more performant.

Reply:
Used `is_validate_assignee_identifier_name` in [f1d4290](https://github.com/rolldown/rolldown/pull/2523/commits/f1d42908e7a24bd06536df5d41d40fb377767c0d)